northstar/src/runtime/mount.rs

```diff
@@ -74,29 +77,46 @@ impl MountControl {
     /// Mounts the npk root fs to target and returns the device used to mount (loopback or device mapper)
     pub(super) fn mount(
         &self,
-        npk: Arc<Npk>,
+        npk: &Npk,
         target: &Path,
         key: Option<&PublicKey>,
     ) -> impl Future<Output = Result<MountInfo, Error>> {
-        let key = key.copied();
         let dm = self.dm.clone();
         let lc = self.lc.clone();
+        let key = key.cloned();
         let target = target.to_owned();
+        let fd = npk.as_raw_fd();
+        let fsimg_size = npk.fsimg_size();
+        let fsimg_offset = npk.fsimg_offset();
+        let name = npk.manifest().name.clone();
+        let version = npk.manifest().version.clone();
+        let verity_header = npk.verity_header().cloned();
+        let hashes = npk.hashes().cloned();
 
         async move {
             let start = time::Instant::now();
 
-            debug!(
-                "Mounting {}:{}",
-                npk.manifest().name,
-                npk.manifest().version
-            );
-            let device = mount(dm, lc, &npk, &target, key.is_some()).await?;
+            debug!("Mounting {}:{}", name, version);
+            let device = mount(
+                dm,
+                lc,
+                fd,
+                fsimg_offset,
+                fsimg_size,
+                &name,
+                &version,
+                verity_header,
+                hashes,
+                &target,
+                key.is_some(),
+            )
```

Comment:
Yes. It's because we need to move all the mount attributes into the mount future. If we'd use `&Npk` this would need to be `'static` which is not possible.

Reply:
hmm well that is not very nice :disappointed:  Maybe having `Arc`  in the interface wasn't such a bad idea. How about making `get()` return some `NpkRef` type that implements `Deref<Target = Npk>`?

Reply:
In that case the `NpkRef` would serve the same purpose as an `Arc` because it must implement `Send`. Instead of introducing `NpkRef` I'd rather switch to the `Arc`. Not sure if that would be sufficient for the future.